--- a/web/core/assets/scaffold/files/htaccess
+++ b/web/.htaccess
@@ -1,7 +1,6 @@
 #
 # Apache/PHP/Drupal settings:
 #
-
 # Protect files and directories from prying eyes.
 <FilesMatch "\.(engine|inc|install|make|module|profile|po|sh|.*sql|theme|twig|tpl(\.php)?|xtmpl|yml)(~|\.sw[op]|\.bak|\.orig|\.save)?$|^(\.(?!well-known).*|Entries.*|Repository|Root|Tag|Template|composer\.(json|lock)|web\.config|yarn\.lock|package\.json)$|^#.*#$|\.php(~|\.sw[op]|\.bak|\.orig|\.save)$">
   <IfModule mod_authz_core.c>
@@ -25,6 +24,9 @@ AddEncoding gzip svgz
 # Add correct encoding for webp.
 AddType image/webp .webp

+# Add correct encoding for .txt
+AddType "text/plain; charset=UTF-8" .txt
+
 # Most of the following PHP settings cannot be changed at runtime. See
 # sites/default/default.settings.php and
 # Drupal\Core\DrupalKernel::bootEnvironment() for settings that can be
@@ -63,6 +65,19 @@ AddType image/webp .webp
 <IfModule mod_rewrite.c>
   RewriteEngine on

+  # Redirect robots.txt on dev servers to a "deny all" variant.
+  RewriteCond %{HTTP_HOST} ^(.+)\.((dev\d+|dgm)\.webtourismus\.at|dedi3828\.your-server\.de)$
+  RewriteRule ^robots\.txt$ robots.dev.txt [L,NS]
+
+  # To redirect all users to access the site WITH the secure 'https://' protocol,
+  # (http://www.example.com/foo will be redirected to https://www.example.com/foo)
+  # uncomment the following:
+  RewriteCond %{HTTPS} off
+  RewriteCond %{HTTP:X-Forwarded-Proto} !https
+  RewriteCond %{HTTP_HOST} !\.dedi(103|1661|3828)\.your-server\.de$ [NC]
+  RewriteCond %{REQUEST_FILENAME} !-f
+  RewriteRule ^(.*)$ https://%{HTTP_HOST}%{REQUEST_URI} [L,R=301]
+
   # Set "protossl" to "s" if we were accessed via https://.  This is used later
   # if you enable "www." stripping or enforcement, in order to ensure that
   # you don't bounce between http and https.
@@ -96,9 +111,11 @@ AddType image/webp .webp
   # To redirect all users to access the site WITH the 'www.' prefix,
   # (http://example.com/foo will be redirected to http://www.example.com/foo)
   # uncomment the following:
-  # RewriteCond %{HTTP_HOST} .
-  # RewriteCond %{HTTP_HOST} !^www\. [NC]
-  # RewriteRule ^ http%{ENV:protossl}://www.%{HTTP_HOST}%{REQUEST_URI} [L,R=301]
+  RewriteCond %{HTTP_HOST} .
+  RewriteCond %{HTTP_HOST} !^www\. [NC]
+  RewriteCond %{HTTP_HOST} !\.[^\.]+\.webtourismus\.at$ [NC]
+  RewriteCond %{HTTP_HOST} !\.dedi(103|1661|3828)\.your-server\.de$ [NC]
+  RewriteRule ^ http%{ENV:protossl}://www.%{HTTP_HOST}%{REQUEST_URI} [L,R=301]
   #
   # To redirect all users to access the site WITHOUT the 'www.' prefix,
   # (http://www.example.com/foo will be redirected to http://example.com/foo)
