--- a/web/.htaccess	2025-04-02 23:04:03.000000000 +0200
+++ b/web/.htaccess	2025-05-01 16:03:32.837242595 +0200
@@ -63,6 +63,19 @@
 <IfModule mod_rewrite.c>
   RewriteEngine on

+  # Redirect robots.txt on dev servers to a "deny all" variant.
+  RewriteCond %{HTTP_HOST} ^(.+)\.((dev\d+|dgm)\.webtourismus\.at|dedi3828\.your-server\.de)$
+  RewriteRule ^robots\.txt$ robots.dev.txt [L,NS]
+
+  # To redirect all users to access the site WITH the secure 'https://' protocol,
+  # (http://www.example.com/foo will be redirected to https://www.example.com/foo)
+  # uncomment the following:
+  RewriteCond %{HTTPS} off
+  RewriteCond %{HTTP:X-Forwarded-Proto} !https
+  RewriteCond %{HTTP_HOST} !\.dedi(103|1661|3828)\.your-server\.de$ [NC]
+  RewriteCond %{REQUEST_FILENAME} !-f
+  RewriteRule ^(.*)$ https://%{HTTP_HOST}%{REQUEST_URI} [L,R=301]
+
   # Set "protossl" to "s" if we were accessed via https://.  This is used later
   # if you enable "www." stripping or enforcement, in order to ensure that
   # you don't bounce between http and https.
@@ -96,9 +109,11 @@
   # To redirect all users to access the site WITH the 'www.' prefix,
   # (http://example.com/foo will be redirected to http://www.example.com/foo)
   # uncomment the following:
-  # RewriteCond %{HTTP_HOST} .
-  # RewriteCond %{HTTP_HOST} !^www\. [NC]
-  # RewriteRule ^ http%{ENV:protossl}://www.%{HTTP_HOST}%{REQUEST_URI} [L,R=301]
+  RewriteCond %{HTTP_HOST} .
+  RewriteCond %{HTTP_HOST} !^www\. [NC]
+  RewriteCond %{HTTP_HOST} !\.[^\.]+\.webtourismus\.at$ [NC]
+  RewriteCond %{HTTP_HOST} !\.dedi(103|1661|3828)\.your-server\.de$ [NC]
+  RewriteRule ^ http%{ENV:protossl}://www.%{HTTP_HOST}%{REQUEST_URI} [L,R=301]
   #
   # To redirect all users to access the site WITHOUT the 'www.' prefix,
   # (http://www.example.com/foo will be redirected to http://example.com/foo)
