diff --git a/src/Plugin/Block/CookiesDocsBlock.php b/src/Plugin/Block/CookiesDocsBlock.php
index 23039b7..cececbe 100644
--- a/src/Plugin/Block/CookiesDocsBlock.php
+++ b/src/Plugin/Block/CookiesDocsBlock.php
@@ -155,7 +155,7 @@ class CookiesDocsBlock extends BlockBase implements ContainerFactoryPluginInterf
   /**
    * {@inheritdoc}
    */
-  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state): void {
     $this->configuration['show_cookies_settings_button'] = $form_state->getValue('show_cookies_settings_button');
     parent::submitConfigurationForm($form, $form_state);
   }
