From 78cec6047a861250ae4ef65f9eb69f5947bbc1f6 Mon Sep 17 00:00:00 2001
From: Reinhard Hutter <17421-hudri@users.noreply.drupalcode.org>
Date: Wed, 29 May 2024 08:42:23 +0000
Subject: [PATCH] Remove untranslatable fields message

---
 .../src/ContentTranslationHandler.php                    | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/core/modules/content_translation/src/ContentTranslationHandler.php b/core/modules/content_translation/src/ContentTranslationHandler.php
index b4911b60afa3..de72d6a4bb48 100644
--- a/core/modules/content_translation/src/ContentTranslationHandler.php
+++ b/core/modules/content_translation/src/ContentTranslationHandler.php
@@ -616,7 +616,14 @@ public function entityFormSharedElements($element, FormStateInterface $form_stat
             // Hide widgets for untranslatable fields.
             if ($hide_untranslatable_fields && isset($field_definitions[$key])) {
               $element[$key]['#access'] = FALSE;
-              $display_warning = TRUE;
+              // ################################################################################
+              //
+              // This patch completely removes the irritating message
+              // "Fields that apply to all languages are hidden to avoid conflicting changes"
+              //
+              //@see https://www.drupal.org/project/drupal/issues/3445965
+              //
+              // ################################################################################
             }
           }
           else {
-- 
GitLab

