From 0d28f7468cdc60e06432737744f09461ffa8d743 Mon Sep 17 00:00:00 2001
From: Dieter Holvoet <dieter.holvoet@gmail.com>
Date: Mon, 20 Mar 2023 14:30:45 +0100
Subject: [PATCH 1/3] Integrate taxonomy forms with the Gin sidebar

---
 src/GinContentFormHelper.php | 48 ++++++++++++++++++++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/src/GinContentFormHelper.php b/src/GinContentFormHelper.php
index c1a92f2f8..3c604df65 100644
--- a/src/GinContentFormHelper.php
+++ b/src/GinContentFormHelper.php
@@ -3,6 +3,7 @@
 namespace Drupal\gin;
 
 use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Entity\ContentEntityFormInterface;
 use Drupal\Core\Extension\ModuleHandlerInterface;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Render\Element;
@@ -10,6 +11,7 @@ use Drupal\Core\Routing\RouteMatchInterface;
 use Drupal\Core\Session\AccountInterface;
 use Drupal\Core\StringTranslation\StringTranslationTrait;
 use Drupal\Core\Theme\ThemeManagerInterface;
+use Drupal\taxonomy\TermInterface;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Component\HttpFoundation\RequestStack;
 
@@ -202,6 +204,8 @@ class GinContentFormHelper implements ContainerInjectionInterface {
       return;
     }
 
+    $base_form_id = $form_state->getBuildInfo()['base_form_id'] ?? NULL;
+
     // Provide a default meta form element if not already provided.
     // @see NodeForm::form()
     $form['advanced']['#attributes']['class'][] = 'entity-meta';
@@ -263,6 +267,47 @@ class GinContentFormHelper implements ContainerInjectionInterface {
       ];
     }
 
+    // Taxonomy term forms.
+    if ($base_form_id === 'taxonomy_term_form') {
+      $form_object = $form_state->getFormObject();
+      assert($form_object instanceof ContentEntityFormInterface);
+
+      $term = $form_object->getEntity();
+      assert($term instanceof TermInterface);
+
+      if ($form['langcode']) {
+        $form['langcode']['widget'][0]['#type'] = 'details';
+        $form['langcode']['widget'][0]['#group'] = 'advanced';
+      }
+
+      // Move pathauto into sidebar. We have to copy this code from the PathWidget
+      // because, at the time the PathWidget is processed the advanced group is
+      // still not present. The group will be created later in a form alter.
+      $form['path_settings'] = [
+        '#type' => 'details',
+        '#title' => t('URL path settings'),
+        '#open' => !empty($form['path']['widget'][0]['alias']['#value']),
+        '#group' => 'advanced',
+        '#access' => !empty($form['path']['#access']) && $term->hasField('path') && $term->get('path')->access('edit'),
+        '#attributes' => [
+          'class' => ['path-form'],
+        ],
+        '#attached' => [
+          'library' => ['path/drupal.path'],
+        ],
+        '#weight' => 30,
+      ];
+      $form['path']['#group'] = 'path_settings';
+
+      if ($form['content_translation']) {
+        $form['content_translation']['#group'] = 'advanced';
+      }
+
+      if ($form['relations']) {
+        $form['relations']['#group'] = 'advanced';
+      }
+    }
+
     // Specify necessary node form theme and library.
     // @see claro_form_node_form_alter
     $form['#theme'] = ['node_edit_form'];
@@ -379,6 +424,9 @@ class GinContentFormHelper implements ContainerInjectionInterface {
       'entity.media.canonical',
       'entity.node.content_translation_add',
       'entity.node.content_translation_edit',
+      'entity.taxonomy_term.edit_form',
+      'entity.taxonomy_term.add_form',
+      'entity.taxonomy_term.content_translation_add',
       'quick_node_clone.node.quick_clone',
       'entity.node.edit_form',
     ];
-- 
GitLab


From 37f75f0018dbed2ac573684ec0738d0c5d77ab7e Mon Sep 17 00:00:00 2001
From: Dieter Holvoet <dieter.holvoet@gmail.com>
Date: Mon, 10 Jul 2023 14:06:58 +0200
Subject: [PATCH 2/3] Fix warnings if certain fields are hidden

---
 src/GinContentFormHelper.php | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/GinContentFormHelper.php b/src/GinContentFormHelper.php
index 3c604df65..4b1546157 100644
--- a/src/GinContentFormHelper.php
+++ b/src/GinContentFormHelper.php
@@ -275,7 +275,7 @@ class GinContentFormHelper implements ContainerInjectionInterface {
       $term = $form_object->getEntity();
       assert($term instanceof TermInterface);
 
-      if ($form['langcode']) {
+      if (isset($form['langcode'])) {
         $form['langcode']['widget'][0]['#type'] = 'details';
         $form['langcode']['widget'][0]['#group'] = 'advanced';
       }
@@ -299,11 +299,11 @@ class GinContentFormHelper implements ContainerInjectionInterface {
       ];
       $form['path']['#group'] = 'path_settings';
 
-      if ($form['content_translation']) {
+      if (isset($form['content_translation'])) {
         $form['content_translation']['#group'] = 'advanced';
       }
 
-      if ($form['relations']) {
+      if (isset($form['relations'])) {
         $form['relations']['#group'] = 'advanced';
       }
     }
-- 
GitLab


From 4925cd4aa6153d278e9ba925eb1d515183392ba5 Mon Sep 17 00:00:00 2001
From: Mark Jones <mark@justanothermark.co.uk>
Date: Wed, 20 Sep 2023 11:53:01 +0100
Subject: [PATCH 3/3] Undo langcode field movement and added missing
 translation route.

---
 src/GinContentFormHelper.php | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/GinContentFormHelper.php b/src/GinContentFormHelper.php
index 4b1546157..aca6fb0b7 100644
--- a/src/GinContentFormHelper.php
+++ b/src/GinContentFormHelper.php
@@ -275,11 +275,6 @@ class GinContentFormHelper implements ContainerInjectionInterface {
       $term = $form_object->getEntity();
       assert($term instanceof TermInterface);
 
-      if (isset($form['langcode'])) {
-        $form['langcode']['widget'][0]['#type'] = 'details';
-        $form['langcode']['widget'][0]['#group'] = 'advanced';
-      }
-
       // Move pathauto into sidebar. We have to copy this code from the PathWidget
       // because, at the time the PathWidget is processed the advanced group is
       // still not present. The group will be created later in a form alter.
@@ -427,6 +422,7 @@ class GinContentFormHelper implements ContainerInjectionInterface {
       'entity.taxonomy_term.edit_form',
       'entity.taxonomy_term.add_form',
       'entity.taxonomy_term.content_translation_add',
+      'entity.taxonomy_term.content_translation_edit',
       'quick_node_clone.node.quick_clone',
       'entity.node.edit_form',
     ];
-- 
GitLab

